lib/streamlit/elements/exception.py

```diff
@@ -131,10 +147,40 @@ def marshall(exception_proto: ExceptionProto, exception: BaseException) -> None:
         )
 
     if is_uncaught_app_exception:
-        uae = cast(UncaughtAppException, exception)
-        exception_proto.message = _GENERIC_UNCAUGHT_EXCEPTION_TEXT
-        type_str = str(type(uae.exc))
-        exception_proto.type = type_str.replace("<class '", "").replace("'>", "")
+        show_error_details = config.get_option("client.showErrorDetails")
+
+        # Options for show error details config.
+        FULL = "full"
```

Comment:
I am wondering whether we should use an `Enum` class here like we do for example for [ColumnDateKind](https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/elements/lib/column_config_utils.py#L47) to make the typing nicer / more structured. What do you think?